blogposts/2021/how-to-deconstruct-the-monolith.md

```diff
@@ -0,0 +1,106 @@
+---
+title: 'How to deconstruct the monolith'
+author: Marcelo Oliveira
+authorUrl: 
+description: ''
+heroImage: https://sourcegraphstatic.com/blog/deconstructing-the-monolith.gif
+socialImage: https://sourcegraphstatic.com/blog/deconstructing-the-monolith.gif
+publishDate: 2021-06-16T18:00-07:00
+tags: [blog]
+slug: how-to-deconstruct-the-monolith
+published: true
+---
+
+![deconstructing the monolith gif](https://sourcegraphstatic.com/blog/deconstructing-the-monolith.gif)
+
+[Monoliths](https://en.wikipedia.org/wiki/Monolithic_application) are the traditional way of building applications. A monolith starts as a small piece of software that grows over time with new features. Unfortunately, this all-in-one software approach makes it difficult to scale our applications. Even when we build a monolith as a well-organized structure, our application’s codebase can become so large that individual developers and teams don’t understand how it all fits together.
+
+In contrast, [microservices](https://en.wikipedia.org/wiki/Microservices) make scalability possible by breaking the application into independent yet integrated components and services. In this architecture, a system can increase or decrease performance and cost by adjusting the resources allocated to the components. With microservices, we can handle the number of instances allocated to each service as needed automatically. This is done using a container orchestration system, such as [Kubernetes](https://kubernetes.io/).
+
+In a distributed system, multiple teams can independently develop different features in separate services. When successfully implemented, a microservices development culture enables each team to be autonomous, requiring minimal coordination to work with other teams. As a result, it boasts improved code quality and long-term code ownership.
+
+Let’s look at some steps you can take to deconstruct a monolith gradually and replace it with a microservices application.
```

Comment:
I think we can draw the connection better here by adding something like "This is all great if you build your application with microservices from the outset, but what about existing monolithic applications?" (not wedded to this wording, but you get the idea)

Reply:
Great point! I think your connecting sentence works.